### PR TITLE
Add synchronozation to list operations

### LINKED ISSUE #38: Rare ArrayIndexOutOfBoundsException on multithreaded runs
Caught on http://ci.qatools.ru/job/allure-cucumber-jvm-adaptor_master-deploy/8/console
Never reproduced myself but to be sure what all safe need to add synchronizet to all ops with plugin private lists

### PR DESCRIPTION
**Pull request by clicman**
Fixes #38

### CODE REVIEW

File: src/main/java/ru/yandex/qatools/allure/cucumberjvm/AllureReporter.java [hunk 1]
```diff
@@ -2,13 +2,7 @@
...
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Field;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Objects;
-import java.util.UUID;
+import java.util.*;
```
Comment (baev) [inline src/main/java/ru/yandex/qatools/allure/cucumberjvm/AllureReporter.java]: remove this change

**Review by baev — APPROVED** [review]

**Comment by baev** [comment]
Is there any way to write the test to reproduce the problem?

**Comment by baev** [comment]
This test found the same problem in allure1 https://github.com/allure-framework/allure1/blob/master/allure-java-adaptor-api/src/test/java/ru/yandex/qatools/allure/AllureLifecycleTest.java#L186-L220

**Comment by clicman** [comment]
There shouldn't be reproducable, because cucumber-jvm test supports *classes* only threading. And in this case for each class new Plugin instance initiated. So I have no idea how it possible.

**Comment by baev** [comment]
Hm.. Is it possible to manually parallelize steps during test?

**Comment by clicman** [comment]
Yes, it possible though pointing `<parallel>methods</parallel>` in surefire config, but it will crash cucumber-jvm itself on its not synchronized internal lists and it will happen before original problem. As I said cucumber-jvm supports classes parallelization only.

**Comment by baev** [comment]
I got it, thanks.

**Pull request merged**